app/models/payment.rb

```diff
@@ -12,19 +12,28 @@ class Payment < ApplicationRecord
   has_many :refunds
   has_many :integration_resources, as: :syncable
 
-  PAYMENT_TYPES = {provider: "provider", manual: "manual"}
+  PAYMENT_TYPES = {provider: 'provider', manual: 'manual'}.freeze
   attribute :payment_type, :string
   enum :payment_type, PAYMENT_TYPES, default: :provider, prefix: :payment_type
   validates :payment_type, presence: true
   validates :reference, presence: true, length: {maximum: 40}, if: -> { payment_type_manual? }
   validates :reference, absence: true, if: -> { payment_type_provider? }
+  validate :max_invoice_paid_amount_cents, on: :create
+  validate :payment_request_succeeded, on: :create
 
   delegate :customer, to: :payable
 
   enum payable_payment_status: PAYABLE_PAYMENT_STATUS.map { |s| [s, s] }.to_h
 
-  validate :max_invoice_paid_amount_cents, on: :create
-  validate :payment_request_succeeded, on: :create
+  scope :for_organization, lambda { |organization|
+    invoices_join = ActiveRecord::Base.sanitize_sql_array(
+      ["LEFT JOIN invoices AS i ON i.id = payments.payable_id AND payments.payable_type = 'Invoice' AND i.organization_id = ?", organization.id]
+    )
+    payment_requests_join = ActiveRecord::Base.sanitize_sql_array(
+      ["LEFT JOIN payment_requests AS pr ON pr.id = payments.payable_id AND payments.payable_type = 'PaymentRequest' AND pr.organization_id = ?", organization.id]
+    )
+    joins(invoices_join).joins(payment_requests_join).where('i.id IS NOT NULL OR pr.id IS NOT NULL')
+  }
```

Comment:
this scope is a little bit hard to understand at first, I would suggest something like this to improve the code clarity...

```
scope :for_organization, lambda { |organization|
  joins(:payable).where(
    <<~SQL, org_id: organization.id
      (payments.payable_type = 'Invoice' AND invoices.organization_id = :org_id) OR
      (payments.payable_type = 'PaymentRequest' AND payment_requests.organization_id = :org_id)
    SQL
  )
}
```

what do you think?